packages/rest/src/providers/parse-params.provider.ts

```diff
@@ -17,12 +22,18 @@ import {Request, ParseParams, RequestBodyParserOptions} from '../types';
  * @returns {ParseParams} The handler function that will parse request args.
  */
 export class ParseParamsProvider implements Provider<ParseParams> {
+  private requestBodyParser: RequestBodyParser;
+
   constructor(
     @inject(RestBindings.REQUEST_BODY_PARSER_OPTIONS, {optional: true})
-    private options?: RequestBodyParserOptions,
-  ) {}
+    options?: RequestBodyParserOptions,
+    @inject.tag(REQUEST_BODY_PARSER_TAG, {optional: true})
+    parsers: BodyParser[] = [],
+  ) {
+    this.requestBodyParser = new RequestBodyParser(options, parsers);
```

Comment:
Hang on. Why are we instantiating BodyParser directly when the class is already wired up for dependency injection?

IMO, we should receive `RequestBodyParser` via dependency injection. I suspect this may simplify unit testing?

Alternatively, we can consider `RequestBodyParser` as an implementation detail that is not customizable, in which case I don't see why should RequestBodyParser be decorated for dependency injection.

Let's discuss.

Reply:
Switching to DI.